web/src/components/profile/FreezeAccount.tsx

```diff
@@ -0,0 +1,59 @@
+import { FC, useState } from 'react';
+import { Button } from 'web/src/components/ui/Button';
+import { useT } from 'web/src/hooks/useT';
+import { Modal, ModalHeader } from 'web/src/components/ui/Modal';
+import { Box } from 'web/src/components/ui/Box';
+import { useFreezeProfile } from 'web/src/hooks/mutations/useUpdateProfile';
+import { FreezeAccountContent } from './FreezeAccountСontent';
+
+interface Props {
+  isSelfFrozen: boolean;
+}
+
+export const FreezeAccount: FC<Props> = ({ isSelfFrozen }) => {
+  const t = useT();
+  const [open, setOpen] = useState(false);
+  const { freezeAccount, status, reset } = useFreezeProfile();
+
+  const handleClick = () => {
+    reset();
+    setOpen(true);
+  };
+
+  const handleClose = () => setOpen(false);
+  const handleClickFreeze = () => freezeAccount();
+
+  return (
+    <>
+      <Button color="secondary" disabled={isSelfFrozen} onClick={handleClick}>
+        {t('Freeze Account')}
+      </Button>
+      <Modal size="lg" show={open} onClose={handleClose}>
+        <ModalHeader>{t('Freeze Account')}</ModalHeader>
+        <Box
+          fontSize="medium"
+          mx="6x"
+          py="4x"
+          borderTopWidth="1x"
+          borderColor="modalHeaderBorderBottom"
+          borderTopStyle="solid"
+        >
+          <FreezeAccountContent status={status} />
+        </Box>
+        <Box display="flex" alignItems="center" px="6x" py="4x">
+          {status !== 'running' && (
+            <>
+              <Button color="secondary" fullWidth onClick={handleClickFreeze}>
+                {t('Freeze')}
+              </Button>
+              <Box w="4x" flexShrink={0} />
+            </>
+          )}
```

Comment:
```suggestion
              <Button color="secondary" fullWidth disabled={status === 'running'} onClick={handleClickFreeze}>
                {t('Freeze')}
              </Button>
              <Box w="4x" flexShrink={0} />
```
лучше просто кнопку блокировать, иначе скачет кнопка "закрыть" не красиво